src/plugins/wizard/public/application/utils/state_management/config_slice.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { IndexPatternField } from '../../../../../data/public';
+
+interface ConfigSections {
+  [id: string]: {
+    title: string;
+    fields: IndexPatternField[];
+  };
+}
+interface ConfigState {
+  configSections: ConfigSections;
+}
+
+// TODO: Temp. Remove once visualizations can be refgistered and editor configs can be passed along
+// TODO: this is a placeholder while the config section is iorned out
+const initialState: ConfigState = {
+  configSections: {
+    x: {
+      title: 'X Axis',
+      fields: [],
+    },
+    y: {
+      title: 'Y Axis',
```

Comment:
Should we have `i18n.translate` for the 2 `title` props?

Reply:
Yes, but this code will be completely replaced by the Contributions PR that is following this. There it is set by the visualization type directly. That being said, all this code is still in flux. Thats why we have a separate issue to track updating all hard coded strings.